### PR TITLE
[34218] Resolve several issues with Project Gantt charts

### PR DESCRIPTION
**Pull request by andersdd**
- Fix version 5.0 compatibility issues by swapping out QWebView with QWebEngineView.
- Resolve cross-source security by embedding source into the HTML.  
- Adjust sizing of chart to fit better in modern monitors. 
- Resolve date issue that was causing the gantt chart bars to be widely incorrect.

I note the package requires 2.5.x to be installed on a version 5.0 xTuple system so these changes are _not_ backwards compatible

### CODE REVIEW

File: extensions/time_expense/foundation-database/te/tables/pkgscript/projectGantt.js [hunk 1]
```diff
@@ -9,9 +9,10 @@
  */
 
 include("xtte");
+include("jsGantt");
 xtte.projectGantt = new Object;
 
-var _webView = new QWebView(mywindow);
+var _webView = new QWebEngineView(mywindow);
```
Comment (gilmoskowitz) [inline extensions/time_expense/foundation-database/te/tables/pkgscript/projectGantt.js]:
```suggestion
var _webView = QWebView ? new QWebView(mywindow) : new QWebEngineView(mywindow);
```

Comment (andersdd) [reply extensions/time_expense/foundation-database/te/tables/pkgscript/projectGantt.js]: Has the entire package been checked for compatibility?   Currently the 2.5.x package.xml is set to support xTuple 5.0.0-alpha + only which indicates there might be conflicts.  Do you want me to go over the whole xtte package to suport pre/post v5.0?

Comment (gilmoskowitz) [reply extensions/time_expense/foundation-database/te/tables/pkgscript/projectGantt.js]: Please do check for pre/post 5.0. My expectation is that it's close.
As for the suggested code, let me guess — `QWebView is not declared` or some such... In that case `var QWebView;` first, then see if it's truthy.

Comment (andersdd) [reply extensions/time_expense/foundation-database/te/tables/pkgscript/projectGantt.js]: Yes `QWebView` not declared.  I've done a version check and pushed the fix here.  Doing a code review and already found some v5.0 specific code so going to have to go through more thoroughly.  Will post to a separate PR

**Pull request merged**